File: CS/Scaffolding.ValidationErrors/DepartmentContextDataModel/DesignTimeDepartmentContextUnitOfWork.cs
using DevExpress.Mvvm.DataModel;
using DevExpress.Mvvm.DataModel.DesignTime;
using Scaffolding.ValidationErrors.Model;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Scaffolding.ValidationErrors.DepartmentContextDataModel {

    /// <summary>
    /// A DepartmentContextDesignTimeUnitOfWork instance that represents the design-time implementation of the IDepartmentContextUnitOfWork interface.
    /// </summary>
    public class DepartmentContextDesignTimeUnitOfWork : DesignTimeUnitOfWork, IDepartmentContextUnitOfWork {

        /// <summary>
        /// Initializes a new instance of the DepartmentContextDesignTimeUnitOfWork class.
        /// </summary>
        public DepartmentContextDesignTimeUnitOfWork() {
        }

        IRepository<Department, int> IDepartmentContextUnitOfWork.Departments
        {
            get { return GetRepository((Department x) => x.DepartmentID); }
        }
    }
}

File: VB/Scaffolding.ValidationErrors/DepartmentContextDataModel/DesignTimeDepartmentContextUnitOfWork.vb
Imports DevExpress.Mvvm.DataModel
Imports DevExpress.Mvvm.DataModel.DesignTime
Imports Scaffolding.ValidationErrors.Model
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace Scaffolding.ValidationErrors.DepartmentContextDataModel

	''' <summary>
	''' A DepartmentContextDesignTimeUnitOfWork instance that represents the design-time implementation of the IDepartmentContextUnitOfWork interface.
	''' </summary>
	Public Class DepartmentContextDesignTimeUnitOfWork
		Inherits DesignTimeUnitOfWork
		Implements IDepartmentContextUnitOfWork

		''' <summary>
		''' Initializes a new instance of the DepartmentContextDesignTimeUnitOfWork class.
		''' </summary>
		Public Sub New()
		End Sub

		Private ReadOnly Property IDepartmentContextUnitOfWork_Departments() As IRepository(Of Department, Integer) Implements IDepartmentContextUnitOfWork.Departments
			Get
				Return GetRepository(Function(x As Department) x.DepartmentID)
			End Get
		End Property
	End Class
End Namespace
